crates/raw-vaas/src/payloads/core/gov.rs

```diff
@@ -1 +1,749 @@
+            0x77, 0xd2, 0x39, 0x38, 0xf4, 0xdb, 0xeb, 0x4e, 0x04, 0xc5, 0xc6, 0x0d, 0xe7, 0x72,
+            0x0f, 0x1f, 0x97, 0x65, 0xaa, 0x4a, 0x1d, 0x1a, 0x9a, 0x9b, 0x0c, 0x32, 0xa0, 0xb8,
+            0xc3, 0x59, 0xb1, 0x29, 0x16, 0xe6, 0x7c, 0xb5, 0x10, 0x19, 0x2c, 0x1f, 0xd8, 0xa0,
+            0x9d, 0x21, 0xef, 0x20, 0x1a, 0xe4, 0xdb, 0x89, 0x61, 0xc0, 0x5f, 0xfe, 0xba, 0xc5,
+            0x33, 0x01, 0x11, 0x94, 0xd7, 0xff, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x04, 0x00, 0x00, 0x00, 0x00, 0x00, 0x0f, 0x78, 0xf2, 0x01, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x43, 0x6f, 0x72,
+            0x65, 0x02, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0x02, 0xbe, 0xfa, 0x42, 0x9d, 0x57,
+            0xcd, 0x18, 0xb7, 0xf8, 0xa4, 0xd9, 0x1a, 0x2d, 0xa9, 0xab, 0x4a, 0xf0, 0x5d, 0x0f,
+            0xbe, 0x88, 0xd7, 0xd8, 0xb3, 0x2a, 0x91, 0x05, 0xd2, 0x28, 0x10, 0x0e, 0x72, 0xdf,
+            0xfe, 0x2f, 0xae, 0x07, 0x05, 0xd3, 0x1c,
+        ];
+
+        let raw_vaa = Vaa::parse(vaa.as_slice()).unwrap();
+        assert_eq!(raw_vaa.version(), 1);
+        assert_eq!(raw_vaa.guardian_set_index(), 0);
+        assert_eq!(raw_vaa.signature_count(), 1);
+
+        let body = raw_vaa.body();
+        assert_eq!(body.timestamp(), 294967295);
+        assert_eq!(body.nonce(), 0);
+        assert_eq!(body.emitter_chain(), 1);
+
+        let payload = CoreBridgeGovPayload::try_from(raw_vaa.payload()).unwrap();
+
+        let guardian_set_update = payload.decree().guardian_set_update().unwrap();
+
+        assert_eq!(guardian_set_update.new_index(), 1);
+        assert_eq!(guardian_set_update.num_guardians(), 2);
+
+        let expected_guardians = [
+            [
+                0xbe, 0xfa, 0x42, 0x9d, 0x57, 0xcd, 0x18, 0xb7, 0xf8, 0xa4, 0xd9, 0x1a, 0x2d, 0xa9,
+                0xab, 0x4a, 0xf0, 0x5d, 0x0f, 0xbe,
+            ],
+            [
+                0x88, 0xd7, 0xd8, 0xb3, 0x2a, 0x91, 0x05, 0xd2, 0x28, 0x10, 0x0e, 0x72, 0xdf, 0xfe,
+                0x2f, 0xae, 0x07, 0x05, 0xd3, 0x1c,
+            ],
+        ];
+        for (i, expected) in expected_guardians.iter().enumerate() {
+            let guardian = guardian_set_update.guardian_at(i);
+            assert_eq!(guardian, *expected);
+        }
+
+        // Try to access out of bounds.
+        let err = guardian_set_update.try_guardian_at(2).err().unwrap();
+        assert_eq!(err, "Exceeds number of encoded guardians");
+    }
+
+    #[test]
+    fn invalid_guardian_set_update() {
+        let vaa = [
+            0x01, 0x00, 0x00, 0x00, 0x00, 0x01, 0x00, 0xf2, 0xbf, 0x6e, 0x30, 0xb3, 0xf4, 0x57,
```

Comment:
can this use `hex_literal::hex!` ?

Reply:
Hex literal is nice. Added it here: 67e523eb3d93bc730c00b8e491d617e4a8fd5bc9